runtime/near-vm-logic/src/tests/context.rs

```diff
@@ -104,3 +106,27 @@ decl_test_u128!(
     create_context().account_locked_balance
 );
 decl_test_u128!(test_attached_deposit, attached_deposit, create_context().attached_deposit);
+
+#[test]
+fn test_attached_deposit_view() {
+    #[track_caller]
+    fn test_view(amount: u128) {
+        let mut logic_builder = VMLogicBuilder::default();
+        let mut context = create_context();
+        context.view_config =
+            Some(ViewConfig { max_gas_burnt: VMLimitConfig::test().max_gas_burnt });
+        context.account_balance = 0;
+        context.attached_deposit = amount;
```

Comment:
This is a bit o of a self-defeting tests. I think what we *want* to check here is that, for view calls, we indeed set the `deposit` to zero. I don't know off the top of my head where such a test should live at, could take a look once we are sure we want this.

Reply:
This test just checks that the attached deposit is returned based on context with view config set (this test fails on master).

The assertion of "attached deposit must always be zero with view config context" seemed intuitively out of scope of this PR. If there was some test setup that I missed that would allow checking this, happy to make changes, but it seems most unit tests ignore this variant and construct contexts with a deposit in view context.

Question (to anyone): will there ever be a use case to configure the context for views, or do we want this to always return zero in view?

Alternatively could just statically return 0 in view contexts for `attached_deposit` if the latter is true and we want to be safe